prover/drivers/unit-tests.md

```diff
@@ -2,49 +2,44 @@ This file contains infrastructure for writing tests for K functions.
 
 ```k
 requires "prover.k"
+requires "t/unit/smt.k"
+requires "t/unit/match-assoc.k"
+requires "t/unit/match-assoc-comm.k"
 ```
 
 ```k
-module UNIT-TESTS-SYNTAX
-    imports TOKENS-SYNTAX
-    imports UNIT-TESTS
-    imports DRIVER-KORE-SYNTAX
-```
-
-When K generates the grammar for parsing rules in modules, it adds a production
-for every sort into the KItem sort. This is *not* done for the program parsing
-grammar. The consequence of this is that `Pattern`s etc will not be parsed
-as keys/values in `Map`s, since `Map` uses `KItem` directly in it's productions.
-To work around this, we manually add such productions below.
-
-```k
-    syntax KItem ::= Pattern
-                   | MatchResults
-                   | CheckSATResult
-    syntax K ::= ".K" [klabel(#EmptyK), symbol]
-```
-
-```k
+module DRIVER-UNIT-TEST
+  imports TEST-CHECKSAT
+  imports TEST-MATCH-ASSOC
+  imports TEST-MATCH-ASSOC-COMM
 endmodule
 ```
 
 ```k
-module UNIT-TESTS
-  imports DRIVER-SMT
+module UNIT-TEST
   imports DRIVER-KORE
-  imports STRATEGY-SMT
-  imports STRATEGY-SEARCH-BOUND
-  imports STRATEGY-SIMPLIFICATION
-  imports STRATEGY-MATCHING
-  imports STRATEGY-APPLY-EQUATION
-  imports STRATEGY-UNFOLDING
-  imports STRATEGY-KNASTER-TARSKI
-  imports HEATCOOL-RULES
+  
```

Comment:
This is all (obviously) very hacky, though the important thing is it's nice to finally have unit tests go through. Just trying to think of easy ways to decrease the hackiness, is there a way to parameterize the max number of tests in a test file? Instead of 20 max tests for each file. If it's not simple don't worry about it though.

Reply:
I'm a bit worried about making it parameterizing it, and then forgetting to update the value in the test file when we add a new test. Yes, I think we need to iterate from here, but its a working starting point.

Reply:
We could also make it a hundred. It would be slightly slower, but I don't think it would add more than a couple of seconds.